### ISSUE TITLE
Shadowhawk DLC breaks 1.9.1 loading

### ISSUE THREAD

**Issue by IceRaptor** [BattletechModders/ModTek#172]
Reported on the RT server in #ticket-3186 - users with the ShadowHawk pre-order DLC report various issues when MechEngineer loads. This presents as an 'overweight' issue in the mechlab. 

Reproduction steps:

1. On __vanilla __ 1.9.1 (NO MODS) customize the SHD-2D UMBRA unit
2. Place that unit in a custom lance
3. Play at least one skirmish game with the customized SHD-2D, and save the game.
4. Add a mod that overwrites the SHD-2D mechdef and uses ME; BTA and RT both have overrides
5. Load the game and go to mechbay; SHD-2D should be overweight

Errors will be in cleaned_output_log.txt like:

```
2020-03-10T18:13:00 FYLS [ERROR] System.NullReferenceException: Object reference not set to an instance of an object
  at MechEngineer.Features.Engines.Engine.CalculateStats () [0x0000d] in <3ba2b5bf070a4de58f86bf83eb88b890>:0 
  at MechEngineer.Features.Engines.Engine.set_CoolingDef (MechEngineer.Features.Engines.CoolingDef value) [0x00035] in <3ba2b5bf070a4de58f86bf83eb88b890>:0 
  at MechEngineer.Features.Engines.Engine..ctor (MechEngineer.Features.Engines.CoolingDef coolingDef, MechEngineer.Features.Engines.EngineHeatBlockDef engineEngineHeatBlockDef, MechEngineer.Features.Engines.EngineCoreDef coreDef, 
```

**Comment by IceRaptor** [comment]
Error seems to stem from a race condition between ModTek applying it's harmony patches, and the various singletons that HBS uses to create the DataManagers. Patches on DataManagerUnitInstance.Awake() never fire, and the ModTek patch in VersionManifestUtilities::LoadDefaultManifest never gets invoked for the DataManager associated with the UnityGameInstance. Note the LoadDefaultManifest patches *do* fire when the UnityGameInstance::OnModsInitComplete call fires, which triggers CalcGameHash - which ends up creating a secondary BattleTechResourceLocator (BTRL). So we know the patches work as expected.

ModTek expects be able to intercept the LoadDefaultManifest call from the BattleTechResourceLocator ctor, to add any merged/replaced files with the cache it builds during startup. Because the BTRL being used has only the vanilla files, the SHD-2D mechdef loads all the vanilla items as part of it's dependencies fetch - which is what screws up ME.

**Comment by IceRaptor** [comment]
I've patched UIManager::Awake to force call of BattleTechResourceLocator::RefreshTypedEntires. This gives ModTek the chance to inject it's own cached manifest into the DataManager.ResourceLocator which is then used to load. UIManager::Awake was chosen because in testing I could see it was reliably the first invocation that hit BTRL::EntryByID. It gets invoked from UIManager::Awake()'s instantiation of the FaderController. 

Trying to patch other places later in the load-order never had the desired effect, as they seemed to rely upon the original baseManifest for the SHD-2D load. Only forcing a refresh at the first patchable point seemed to help.

**Comment by IceRaptor** [comment]
Fixed in 0.7.7.3
